lib/auth/auth_with_roles.go

```diff
@@ -2377,14 +2377,19 @@ func (a *ServerWithRoles) GetSessionRecordingConfig(ctx context.Context, opts ..
 }
 
 // SetSessionRecordingConfig sets session recording configuration.
-func (a *ServerWithRoles) SetSessionRecordingConfig(ctx context.Context, recConfig types.SessionRecordingConfig) error {
-	if err := a.action(apidefaults.Namespace, types.KindSessionRecordingConfig, types.VerbCreate); err != nil {
+func (a *ServerWithRoles) SetSessionRecordingConfig(ctx context.Context, newRecConfig types.SessionRecordingConfig) error {
+	storedRecConfig, err := a.authServer.GetSessionRecordingConfig(ctx)
+	if err != nil {
 		return trace.Wrap(err)
 	}
-	if err := a.action(apidefaults.Namespace, types.KindSessionRecordingConfig, types.VerbUpdate); err != nil {
-		return trace.Wrap(err)
+
+	for _, verb := range verbsToReplaceResourceWithOrigin(storedRecConfig) {
```

Comment:
To clarify, this PR allows updating a dynamic config with just `VerbUpdate` and without `VerbCreate`.
Nothing is changed for static (config file) config?

Reply:
Yes, it's the same thing as for the other dynamically configurable resources, I just forgot to change this verb check when upgrading `SessionRecordingConfig` to dynamically configurable.

AFAICT RBAC verbs are never checked for static configuration as the auth init doesn't operate on a `WithRoles` API.